beacon-chain/chaintest/state-replay/main.go

```diff
@@ -0,0 +1,151 @@
+	"github.com/prysmaticlabs/prysm/beacon-chain/operations"
+	"github.com/prysmaticlabs/prysm/beacon-chain/powchain"
+	"github.com/prysmaticlabs/prysm/shared/featureconfig"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	"github.com/sirupsen/logrus"
+	"github.com/x-cray/logrus-prefixed-formatter"
+	"os"
+	"time"
+)
+
+var log = logrus.WithField("prefix", "state-replay")
+
+type mockP2P struct{}
+
+func (m *mockP2P) Broadcast(ctx context.Context, msg proto.Message) {}
+
+func main() {
+	var dbPath = flag.String("db-dir", "", "path to bolt.db dir")
+	flag.Parse()
+
+	customFormatter := new(prefixed.TextFormatter)
+	customFormatter.TimestampFormat = "2006-01-02 15:04:05"
+	customFormatter.FullTimestamp = true
+	logrus.SetFormatter(customFormatter)
+
+	readOnlyDB, err := db.NewReadOnlyDB(*dbPath)
+	if err != nil {
+		log.Fatal(err)
+	}
+	beaconDB, err := db.NewDB("/tmp/state-replay-dir")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer os.RemoveAll("/tmp/state-replay-dir")
+	ctx := context.Background()
+	params.UseDemoBeaconConfig()
+	featureconfig.InitFeatureConfig(&featureconfig.FeatureFlagConfig{})
+
+	// Setup a chain service and powchain service.
+	rpcClient, err := gethRPC.Dial("wss://goerli.prylabs.net/websocket")
+	if err != nil {
+		log.Fatalf("Access to PoW chain is required for validator. Unable to connect to Geth node: %v", err)
+	}
+	httpRPCClient, err := gethRPC.Dial("https://goerli.prylabs.net")
+	if err != nil {
+		log.Fatalf("Access to PoW chain is required for validator. Unable to connect to Geth node: %v", err)
+	}
+	powClient := ethclient.NewClient(rpcClient)
+	httpClient := ethclient.NewClient(httpRPCClient)
+	cfg := &powchain.Web3ServiceConfig{
+		Endpoint:        "wss://goerli.prylabs.net/websocket",
+		DepositContract: common.HexToAddress("0x2348E5Db47B910DDC4478CA202DB280AC5790D2A"),
+		Client:          powClient,
+		Reader:          powClient,
+		Logger:          powClient,
+		HTTPLogger: httpClient,
+		BlockFetcher:    powClient,
+		ContractBackend: powClient,
+		BeaconDB:        beaconDB,
+	}
+	web3Service, err := powchain.NewWeb3Service(ctx, cfg)
+	if err != nil {
+		log.Fatal(err)
+	}
+	attService := attestation.NewAttestationService(ctx, &attestation.Config{
+		BeaconDB: beaconDB,
+	})
+	opsService := operations.NewOpsPoolService(ctx, &operations.Config{
+		BeaconDB: beaconDB,
+	})
+	chainService, err := blockchain.NewChainService(ctx, &blockchain.Config{
+		BeaconDB:    beaconDB,
+		P2p: &mockP2P{},
+		OpsPoolService: opsService,
+		AttsService: attService,
+		Web3Service: web3Service,
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	stateInit := make(chan time.Time)
+	stateInitFeed := chainService.StateInitializedFeed()
+	stateInitFeed.Subscribe(stateInit)
+
+	chainService.Start()
+	defer chainService.Stop()
+	opsService.Start()
+	defer opsService.Stop()
+	attService.Start()
+	defer attService.Stop()
+	web3Service.Start()
+	defer web3Service.Stop()
+
+	log.Info("Waiting for chainstart...")
```

Comment:
```suggestion
	log.Info("Waiting for state to be initialized...")
```

What do you think? We are waiting for state initialized feed.